From 62c4b5f114850d39efdca3df3b9ee83cc0f424ff Mon Sep 17 00:00:00 2001
From: Jeremy Koritzinsky <jekoritz@microsoft.com>
Date: Tue, 21 Jan 2025 16:16:49 -0800
Subject: [PATCH 1/2] Use the documented Artifact extension point to add
 artifacts

The Artifact extension point handles symbol packages automatically for source-build, and it automatically gets support for new features like Artifact Visibility (required for https://github.com/dotnet/sdk/pull/46063)
Backport: https://github.com/NuGet/NuGet.Client/pull/6233
---
 eng/Publishing.props | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/eng/Publishing.props b/eng/Publishing.props
index 38810c73428..1175200dd59 100644
--- a/eng/Publishing.props
+++ b/eng/Publishing.props
@@ -1,11 +1,6 @@
 <Project>
-
   <ItemGroup>
-    <ItemsToPushToBlobFeed Include="$(ArtifactsDir)nupkgs/*.nupkg"
-                           IsShipping="true"
-                           UploadPathSegment="nuget-client" />
-    <_SymbolsPackages Include="$(ArtifactsDir)nupkgs/*.symbols.nupkg" />
-    <ItemsToPushToBlobFeed Remove="@(_SymbolsPackages)" Condition="'$(DotNetBuildSourceOnly)' == 'true'" />
+    <Artifact Include="$(ArtifactsDir)nupkgs/*.nupkg"
+              IsShipping="true" />
   </ItemGroup>
-
-</Project>
\ No newline at end of file
+</Project>

From 9072cb03cb5c2958df318f59389547caa5de2c8a Mon Sep 17 00:00:00 2001
From: Jeremy Koritzinsky <jkoritzinsky@gmail.com>
Date: Wed, 22 Jan 2025 10:07:47 -0800
Subject: [PATCH 2/2] Formatting and PR feedback

---
 eng/Publishing.props | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/eng/Publishing.props b/eng/Publishing.props
index 1175200dd59..41f498d4d4a 100644
--- a/eng/Publishing.props
+++ b/eng/Publishing.props
@@ -1,6 +1,8 @@
-<Project>
-  <ItemGroup>
-    <Artifact Include="$(ArtifactsDir)nupkgs/*.nupkg"
-              IsShipping="true" />
-  </ItemGroup>
-</Project>
+<Project>
+
+  <ItemGroup>
+    <Artifact Include="$(ArtifactsDir)nupkgs/*.nupkg"
+              PublishFlatContainer="false" />
+  </ItemGroup>
+  
+</Project>
